tests/unit_tests/grid/test_points_functions.py

```diff
@@ -0,0 +1,413 @@
+                                              [[75.0, 37.5, 0.0], [20.0, 37.5, 0.0], [10.0, 37.5, 0.0]],
+                                              [[75.0, 37.5, 0.0], [20.0, 37.5, 0.0], [10.0, 37.5, 0.0]]]])
+
+    # Act
+    interpolated_points = pf.interpolated_points(basic_regular_grid,
+                                                 cell_kji0 = cell,
+                                                 interpolation_fractions = interpolation_fractions)
+
+    # Assert
+    np.testing.assert_array_almost_equal(interpolated_points, expected_interpolated_points)
+
+
+def test_split_horizons_points(basic_regular_grid):
+    # Arrange
+    expected_split_horizons_points = np.array([[[[[[0., 0., 0.], [100., 0., 0.]], [[0., 50., 0.], [100., 50., 0.]]],
+                                                 [[[100., 0., 0.], [200., 0., 0.]], [[100., 50., 0.], [200., 50.,
+                                                                                                       0.]]]],
+                                                [[[[0., 50., 0.], [100., 50., 0.]], [[0., 100., 0.], [100., 100., 0.]]],
+                                                 [[[100., 50., 0.], [200., 50., 0.]],
+                                                  [[100., 100., 0.], [200., 100., 0.]]]]],
+                                               [[[[[0., 0., 20.], [100., 0., 20.]], [[0., 50., 20.], [100., 50., 20.]]],
+                                                 [[[100., 0., 20.], [200., 0., 20.]],
+                                                  [[100., 50., 20.], [200., 50., 20.]]]],
+                                                [[[[0., 50., 20.], [100., 50., 20.]],
+                                                  [[0., 100., 20.], [100., 100., 20.]]],
+                                                 [[[100., 50., 20.], [200., 50., 20.]],
+                                                  [[100., 100., 20.], [200., 100., 20.]]]]],
+                                               [[[[[0., 0., 40.], [100., 0., 40.]], [[0., 50., 40.], [100., 50., 40.]]],
+                                                 [[[100., 0., 40.], [200., 0., 40.]],
+                                                  [[100., 50., 40.], [200., 50., 40.]]]],
+                                                [[[[0., 50., 40.], [100., 50., 40.]],
+                                                  [[0., 100., 40.], [100., 100., 40.]]],
+                                                 [[[100., 50., 40.], [200., 50., 40.]],
+                                                  [[100., 100., 40.], [200., 100., 40.]]]]]])
+
+    # Act
+    split_horizons_points = pf.split_horizons_points(basic_regular_grid)
+
+    # Assert
+    np.testing.assert_array_almost_equal(split_horizons_points, expected_split_horizons_points)
+
+
+def test_split_horizons_points_s_bend_faulted_grid(s_bend_faulted_grid):
+    # Act
+    split_horizons_points = pf.split_horizons_points(s_bend_faulted_grid)
```

Comment:
Please add a couple more tests using non zero values for (a) ref_k0 and (b) kp.  For the s_bend model, ref_k0 is in the range 0 to 4 I think.  kp may be 0 or 1.